jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpClientTransportDynamic.java

```diff
@@ -129,6 +130,16 @@ private static ClientConnector findClientConnector(ClientConnectionFactory.Info[
             .orElseGet(ClientConnector::new);
     }
 
+    @Override
+    public void setHttpClient(HttpClient client)
+    {
+        super.setHttpClient(client);
+        infos.stream()
+            .filter(info -> info instanceof HttpClient.Aware)
+            .map(HttpClient.Aware.class::cast)
+            .forEach(info -> info.setHttpClient(getHttpClient()));
```

Comment:
`infos` is an insufficient name.  I have to go find its type to understand the code. How about just being verbose `clientConnectionFactoryInfos`

However, as all the infos are beans anyway, why not look for any contained Aware bean and save some code:
```suggestion
        getContainedBeans(Aware.class).forEach(a -> a.setHttpClient(client));
```

Reply:
actuallyu, as getContainerBeans walks the containment tree, you don't have to do this at every level.  You can just do this one at the top level and find all Aware components under HttpClient

Reply:
> infos is an insufficient name. I have to go find its type to understand the code. How about just being verbose clientConnectionFactoryInfos

Why? It is already called `ClientConnectionFactory.Info`. 
We don't duplicate the name of the outer class into inner classes.

Reply:
I'm not suggesting changing the name of the type, only the variable